### PR TITLE
HTML-ify the Dropdown component

### PR DESCRIPTION
**Pull request by tholex**
Improves usability / accessibility of the dropdown component.

Retains existing API for the component, adds label-friendly `id` option.

JP's original edition: https://github.com/Thinkful/ui/pull/315

### CODE REVIEW

File: src/Dropdown/Dropdown.jsx [hunk 1]
```diff
@@ -99,44 +82,25 @@ const Dropdown = React.createClass({
...
+      "tui-dropdown-container",
+      dropdownClasses);
+
+    const selectedInd = this._determineSelectedInd();
+
+    return <div className={dropdownClasses}>
+      <select
+        id={htmlId}
+        onChange={e => this._handleChange(e)}
+        className="tui-dropdown">
```
Comment (ripleyaffect) [inline src/Dropdown/Dropdown.jsx]: needs an extra indent to visually distinguish from the children

File: src/Dropdown/Dropdown.jsx [hunk 2]
```diff
@@ -62,24 +51,18 @@ const Dropdown = React.createClass({
 
     return data.map((item, ind) => {
       return (
-        <p
-          className={cx("dropdown-item", item.className)}
-          id={ind}
+        <option
+          className={cx(item.className)}
```
Comment (ripleyaffect) [inline src/Dropdown/Dropdown.jsx]: no need to call `cx` here, unless you're supporting `item.className` as not a string

File: src/Dropdown/Dropdown.jsx [hunk 3]
```diff
@@ -99,44 +82,25 @@ const Dropdown = React.createClass({
...
+
+    const selectedInd = this._determineSelectedInd();
+
+    return <div className={dropdownClasses}>
+      <select
+        id={htmlId}
+        onChange={e => this._handleChange(e)}
+        className="tui-dropdown">
+        {selectedInd == -1 &&
+          <option selected disabled key="tui-dropdown-default">{defaultDisplay}</option>}
```
Comment (ripleyaffect) [inline src/Dropdown/Dropdown.jsx]: `defaultDisplay` isn't required and there's no default, should have one of those, or this should default to having the first option selected

File: src/Dropdown/Dropdown.jsx (hunk 2 above)
Comment (tholex) [inline src/Dropdown/Dropdown.jsx]: done

File: src/Dropdown/Dropdown.jsx (hunk 1 above)
Comment (tholex) [inline src/Dropdown/Dropdown.jsx]: done.

File: src/Dropdown/Dropdown.jsx (hunk 3 above)
Comment (tholex) [inline src/Dropdown/Dropdown.jsx]: HTML's default behavior selects the first element, it works without adding the `selected` attribute.

Comment (tholex) [inline src/Dropdown/Dropdown.jsx]: http://codepen.io/tholex/pen/99c3ab917e0c275f5dde4fca58519d1d

**Pull request merged**